src/api/v2/transactions.rs

```diff
@@ -16,6 +18,12 @@ pub trait Submit {
 // TODO: Replace this with avail::PairSigner after implementing required traits in avail-subxt
 pub type AvailSigner = PairSigner<AvailConfig, Pair>;
 
+impl From<&AvailSecretKey> for AvailSigner {
+	fn from(value: &AvailSecretKey) -> Self {
+		AvailSigner::new(value.0.clone())
+	}
+}
```

Comment:
It is best to push cloning to user side, so that he could control allocations.
```suggestion
impl From<AvailSecretKey> for AvailSigner {
	fn from(value: AvailSecretKey) -> Self {
		AvailSigner::new(value.0)
	}
}
```

Reply:
What do you mean? `subxt::ext::sp_core::sr25519::Pair` is the same as `AvailSecretKey` as it just wraps `Pair`. Cloning will be the same in the end.